test/module/irohad/ametsuchi/postgres_executor_test.cpp

```diff
@@ -405,6 +405,103 @@ namespace iroha {
       CHECK_ERROR_CODE_AND_MESSAGE(cmd_result, 2, query_args);
     }
 
+    class RemovePeer : public CommandExecutorTest {
+     public:
+      void SetUp() override {
+        CommandExecutorTest::SetUp();
+        peer = makePeer("address",
+                        shared_model::interface::types::PubkeyType{"pubkey"});
+        another_peer = makePeer(
+            "another_address",
+            shared_model::interface::types::PubkeyType{"another_pubkey"});
+        createDefaultRole();
+        createDefaultDomain();
+        createDefaultAccount();
+        CHECK_SUCCESSFUL_RESULT(
+            execute(*mock_command_factory->constructAddPeer(*peer), true));
+      }
+
+      std::unique_ptr<MockPeer> peer, another_peer;
+    };
+
+    /**
+     * @given command
+     * @when trying to remove peer
+     * @then peer is successfully removed
+     */
+    TEST_F(RemovePeer, Valid) {
+      addAllPerms();
+      auto new_peer =
+          makePeer("new_address",
+                   shared_model::interface::types::PubkeyType{"new_pubkey"});
+      CHECK_SUCCESSFUL_RESULT(
+          execute(*mock_command_factory->constructAddPeer(*new_peer), true));
+
+      CHECK_SUCCESSFUL_RESULT(
+          execute(*mock_command_factory->constructRemovePeer(*peer)));
+
+      auto peers = wsv_query->getPeers();
+      ASSERT_TRUE(peers);
+      ASSERT_TRUE(std::find_if(peers->begin(),
+                               peers->end(),
+                               [this](const auto &peer) {
+                                 return this->peer->address() == peer->address()
+                                     and this->peer->pubkey() == peer->pubkey();
+                               })
+                  == peers->end());
+    }
+
+    /**
+     * @given command
+     * @when trying to remove peer without perms
+     * @then peer is not removed
+     */
+    TEST_F(RemovePeer, NoPerms) {
+      auto new_peer =
+          makePeer("new_address",
+                   shared_model::interface::types::PubkeyType{"new_pubkey"});
+      CHECK_SUCCESSFUL_RESULT(
+          execute(*mock_command_factory->constructAddPeer(*new_peer), true));
+      auto cmd_result =
+          execute(*mock_command_factory->constructRemovePeer(*peer));
+
+      std::vector<std::string> query_args{peer->toString()};
+      CHECK_ERROR_CODE_AND_MESSAGE(cmd_result, 2, query_args);
+    }
+
+    /**
+     * @given command
+     * @when trying to remove nonexistent peer
+     * @then peer is not removed
+     */
+    TEST_F(RemovePeer, NoPeer) {
+      addAllPerms();
+      auto new_peer =
+          makePeer("new_address",
+                   shared_model::interface::types::PubkeyType{"new_pubkey"});
+      CHECK_SUCCESSFUL_RESULT(
+          execute(*mock_command_factory->constructAddPeer(*new_peer), true));
```

Comment:
please add a comment that the new peer is needed to exclude any relations to peer number checks (if this is the reason as I guessed)